### PR TITLE
Support a concept of Saas ('service') contexts for PUE licenses | #66462

### PR DESCRIPTION
**Pull request by barryhughes**
Introduces contexts for PUE license checks. In the case of 'components' (default, normal use - ie plugins but potentially also themes) it will attempt to hook into WP update checks, etc. For 'services' (ie, EA) it will not.

[#66462](https://central.tri.be/issues/66462)

### CODE REVIEW

File: src/Tribe/PUE/Checker.php [hunk 1]
```diff
@@ -161,8 +176,9 @@ public function hooks() {
 			add_filter( 'plugins_api', array( $this, 'inject_info' ), 10, 3 );
 
 			// Check for updates when the WP updates are checked and inject our update if needed.
-			// Only add filter if the TRIBE_DISABLE_PUE constant is not set as true.
-			if ( ! defined( 'TRIBE_DISABLE_PUE' ) || TRIBE_DISABLE_PUE !== true ) {
+			// Only add filter if the TRIBE_DISABLE_PUE constant is not set as true and where
+			// the context is not 'service'
+			if ( ( ! defined( 'TRIBE_DISABLE_PUE' ) || TRIBE_DISABLE_PUE !== true ) && 'service' !== $this->context ) {
```
Comment (bordoni) [inline src/Tribe/PUE/Checker.php]: May be Multiple lines?

**Comment by zbtirrell** [comment]
👍

File: src/Tribe/PUE/Checker.php (hunk 1 above)
Comment (barryhughes) [inline src/Tribe/PUE/Checker.php]: It's within our soft wrap limit - but I could?

Comment (bordoni) [inline src/Tribe/PUE/Checker.php]: Ohh didn't realize that. Sounds good. Merge away.

**Pull request merged**